src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts

```diff
@@ -143,14 +145,30 @@ export class TerminalSearchLinkOpener implements ITerminalLinkOpener {
 		}
 		const sanitizedLink = matchLink.replace(/:\d+(:\d+)?$/, '');
 		try {
-			const uri = await this._getExactMatch(sanitizedLink);
-			if (uri) {
+			const result = await this._getExactMatch(sanitizedLink);
+			if (result) {
+				const { uri: resource, type} = result;
+			if (type === 'file' && resource) {
 				return this._localFileOpener.open({
 					text: matchLink,
-					uri,
+					uri: resource,
 					bufferRange: link.bufferRange,
 					type: link.type
 				});
+			} else if (type === 'dir' && resource) {
+				link = {
+					text: matchLink,
+					uri: resource,
+					bufferRange: link.bufferRange,
+					type: link.type
+				};
+				try {
```

Comment:
oh I expected this to throw if the link wasn't in the workspace. looks like I'll have to check this to decide which opener should be used